include/flags.h

```diff
@@ -0,0 +1,29 @@
+#ifndef FLAGS_H
+#define FLAGS_H
+
+#include <iostream>
+#include <map>
+#include <climits>
+
+class Flags {
+private:
+  int m_argc;
```

Comment:
Are you following any particular style guide? If not, I'd recommend choosing one and sticking strictly to it. Also, what does the `m` prefix represent?